packages/synth.queues/src/services/kafka.queue.service.ts

```diff
@@ -0,0 +1,88 @@
+import { MultivestError, PluginManager } from '@applicature/synth.plugin-manager';
+import * as config from 'config';
+import { createReadStream, createWriteStream, KafkaConsumer } from 'node-rdkafka';
+import * as logger from 'winston';
+import { Errors } from '../errors';
+import { Message, Queue } from '../types';
+import { QueueService } from './queue.service';
+
+export class KafkaQueueService extends QueueService {
+    private producer: any;
+    private consumer: any;
+
+    constructor(pluginManager: PluginManager) {
+        super(pluginManager);
+        if (config.has('kafka.consumer')) {
+            const kafkaConsumer = config.get('kafka.global');
+            this.consumer = new KafkaConsumer(kafkaConsumer, {});
+        }
+    }
+    public async createQueue(queueName: string): Promise<Queue> {
+        if (config.has('kafka.consumer')) {
+            this.consumer = createReadStream(config.get('kafka.global'), config.get('kafka.topic'), {
+                topics: queueName
+            });
+            this.consumer.setPollInterval(100);
+        } else {
+            throw new MultivestError(Errors.KAFKA_CONFIG_NOT_FOUND);
+        }
+
+        if (config.has('kafka.producer')) {
+            this.producer = createWriteStream(config.get('kafka.global'), config.get('kafka.topic'), {
+                topics: queueName
+            });
+            this.producer.setPollInterval(100);
+        } else {
+            throw new MultivestError(Errors.KAFKA_CONFIG_NOT_FOUND);
+        }
+        return Promise.resolve({
```

Comment:
The method is async, thus, you don't need to wrap a response in the extra promise